### ISSUE TITLE
Azure builder cleanup on existing resource group does not delete deployment

### ISSUE THREAD

**Issue by riezebosch** [hashicorp/packer#8038]
There's this limitation on a resource group to only have 800 deployments in history: https://docs.microsoft.com/en-us/azure/azure-subscription-service-limits?WT.mc_id=personal-blog-marouill#resource-group-limits

Because packer uses ARM to create the VM there is a deployment created on each run. For an existing resource group (by using `custom_managed_image_resource_group_name`) you run into this error:

```log
resources.DeploymentsClient#Validate: Failure sending request: StatusCode=409 -- Original Error: autorest/azure: Service returned an error. Status=<nil> Code="DeploymentQuotaExceeded" Message="Creating the deployment 'pkrdpd91j6odngc' would exceed the quota of '800'. The current deployment count is '800', please delete some deployments before creating a new one. Please see https://aka.ms/arm-deploy for usage details."
```

**Comment by azr** [comment]
I put this one on 1.5.0 🙂

**Comment by ryanhartje** [comment]
Hi all,

I've attempted to resolve this, however this would be my first PR to packer. Any guidance is appreciated.

**Comment by sylviamoss** [comment]
Hello! Was this issue solved by PR https://github.com/hashicorp/packer/pull/8496?

**Comment by ryanhartje** [comment]
I was able to reproduce the issue from `master`, then applied my changes and reran the same process, and no longer had an abandoned deployment. I believe this to be resolved.

**Comment by nywilken** [comment]
@ryanhartje thanks for the update. Closing this issue as it has been resolved by #8496. If after the release there are still issues please leave a comment and we'll take a look.
